### ISSUE TITLE
Passwords not refreshing in android with Dropbox

### ISSUE THREAD

**Issue by bharathramh92** [buttercup/buttercup-mobile#163]
** Describe the issue you're having ** 
After I add a new password in desktop, the same diff is not applied to mobile. Basically it is not doing an auto refresh. It would be nice to add a refresh button in the app.

** What device are you using **
 Pixel 2

**Comment by perry-mitchell** [comment]
Yes, we don't have reloading on mobile yet. I'll rename this to indicate that. Thanks!

**Comment by working-name** [comment]
Any update on mobile auto/manual refresh? 

It seems there's reloading on non-mobile given your reply, but I'm not sure what we're referring to. The desktop client seems to refresh if you switch from one vault to another. Are we talking about CMD/CTRL + R - aka reloading the whole Electron session?

**Comment by perry-mitchell** [comment]
> Any update on mobile auto/manual refresh?

None yet, no. We haven't yet given it a lot of thought as to how it might work on mobile. Doing it in the background is not really an option as users might stuble into something which changes in the background. Locking the foreground while we fetch an update is also a pain as this could take some time and be a UX killer. Happy to hear suggestions on this.

> It seems there's reloading on non-mobile given your reply

The browser extension auto-reloads vaults after some minutes. The desktop doesn't yet support this either: https://github.com/buttercup/buttercup-desktop/issues/686

> aka reloading the whole Electron session

This wouldn't be a feature that we'd want to promote as it's highly dependent on a dev feature of electron, which we want hidden from end users (the fact that we use Electron is irrelevant for our software).

That being said, having a menu option to force reload contents now would be great!

**Comment by working-name** [comment]
I see, yes I'm with you on the UI lock-up while you refresh, but that would only constitute a problem if the action was _not initiated by the user_. 

I don't know how bcup works once it has decrypted the vault, but once it has, can it not use browserDB/IndexedDB/WebSQL to keep track of changes and use that to update the UI? That way updating of changes could happen in the background with a worker, on a timer. Main issue I'm unaware of is how to securely handle removing that db on expiration or app close (hopefully there's triggers for the app close event), or worse, crashes.

In addition, it may get hairy on saving multiple edits/new items in a short amount of time. How feasible is it to delay sending the new vault back to storage? I'm thinking of how long it might take a 3G user to send the whole vault back to storage ...

**Comment by fanaz** [comment]
Is there currently a way to force a reload of the vault on iOS? I've updated the vault in the Desktop app and want to see the changes in the mobile app.

I've tried the following

- lock/unlock the vault
- close and reopen the app
- restart the whole phone

None of this seems to work: I see always the old state of the vault.

**Comment by perry-mitchell** [comment]
Locking and unlocking the vault will show the **latest contents** as they are on the remote storage. If you're not seeing what you expect, then the other client did not save to remote.

That's the trick with using cloud storage providers like Dropbox etc., the sync is a bit slower and is usually somewhat passive on desktop sync clients.
